core/message-tags-3.3.md

```diff
@@ -0,0 +1,100 @@
+---
+title: IRCv3.3 Message Tags
+layout: spec
+work-in-progress: true
+copyrights:
+  -
+    name: "Kiyoshi Aman"
+    period: 2016
+    email: "kiyoshi.aman@gmail.com"
+  -
+    name: "James Wheare"
+    period: 2016
+    email: "james@irccloud.com"
+---
+## Introduction
+
+This specification adds a new capability for general message tags support and
+a prefix for expressing client-only tags.
+
+## Motivation
+
+Previously, clients were required to negotiate a capability with servers for each
+supported tag. This made tags inappropriate for client-only features. By adding a
+new base capability, this specification allows clients to indicate support for
+receiving any well-formed tag, whether or not it is recognised or used. This also
+frees servers from having to filter messages tags for each individual client
+response.
+
+The client-only tag prefix allows servers to safely relay untrusted client tags,
+keeping them distinct from server-initiated tags that carry verified meaning.
+
+## Architecture
+
+### Capabilities
+
+This specification adds the `message-tags` capability. Clients requesting
+this capability indicate that they are capable of parsing all well-formed tags,
+even if they don't handle them specifically.
+
+Servers advertising this capability indicate that they are capable of handling
+client-only tags.
+
+### Tags
+
+Client-only tags are client-initiated tags that servers MUST attach as-is
+to any relevant event relayed to other clients. A client-only tag is prefixed
+with a plus sign (`+`) and otherwise conforms to the format specified in
+[IRCv3.2 tags](./message-tags-3.2.html).
+
+Client-only tags MUST be relayed on `PRIVMSG` and `NOTICE` events, and
+MAY be relayed on other events.
+
+The expected client behaviour of individual client-only tags SHOULD be defined
+in separate specifications, in the same way as server-initiated tags.
+
+The updated BNF for keys is as follows:
+
+    <key>           ::= [ '+' ] [ <vendor> '/' ] <sequence of letters, digits, hyphens (`-`)>
+
+Individual tag keys MUST only be used a maximum of once per message. Clients
+receiving messages with more than one occurrence of a tag key SHOULD discard all
+but the final occurrence.
```

Comment:
Perhaps we should add a reminder around here that people should use vendor prefixes for proprietary features.